### PR TITLE
DOC: Improve the RANSAC gallery example

### PR DESCRIPTION
**Pull request by sciunto**
## Description

It was reported to me that this example was misleading.

I added part of the function docstring + merged two examples as the 3D is just a small adaptation of the 2D case.

## Checklist
[It's fine to submit PRs which are a work in progress! But before they are merged, all PRs should provide:]
- [ ] Clean style in [the spirit of PEP8](https://www.python.org/dev/peps/pep-0008/)
- [ ] [Docstrings for all functions](https://github.com/numpy/numpy/blob/master/doc/example.py)
- [ ] Gallery example in `./doc/examples` (new features only)
- [ ] Benchmark in `./benchmarks`, if your changes aren't covered by an
  existing benchmark
- [ ] Unit tests

[For detailed information on these and other aspects see [scikit-image contribution guidelines](https://scikit-image.org/docs/dev/contribute.html)]

## References
[If this is a bug-fix or enhancement, it closes issue # ]
[If this is a new feature, it implements the following paper: ]

## For reviewers

(Don't remove the checklist below.)

- [ ] Check that the PR title is short, concise, and will make sense 1 year
  later.
- [ ] Check that new functions are imported in corresponding `__init__.py`.
- [ ] Check that new features, API changes, and deprecations are mentioned in
      `doc/release/release_dev.rst`.
- [ ] Consider backporting the PR with `@meeseeksdev backport to v0.14.x`

### CODE REVIEW

File: doc/examples/transform/plot_ransac.py [hunk 1]
```diff
@@ -4,7 +4,32 @@
 =========================================
 
 In this example we see how to robustly fit a line model to faulty data using
-the RANSAC algorithm.
+the RANSAC (random sample consensus) algorithm.
+
+First, data are generated from a line on which a gaussian noise is added.
+Faulty points are also added to the data set.
```
Comment (jni) [inline doc/examples/transform/plot_ransac.py]:
```suggestion
Outlier points are also added to the data set.
```

File: doc/examples/transform/plot_ransac.py [hunk 2]
```diff
@@ -54,3 +79,39 @@
 ax.plot(line_x, line_y_robust, '-b', label='Robust line model')
 ax.legend(loc='lower left')
 plt.show()
+
+
+######################################################################
+# The former example is generalized to 3D.
```
Comment (jni) [inline doc/examples/transform/plot_ransac.py]:
```suggestion
# Now, we generalize this example to 3D points.
```

File: doc/examples/transform/plot_ransac.py [hunk 3]
```diff
@@ -4,7 +4,32 @@
...
+the RANSAC (random sample consensus) algorithm.
+
+First, data are generated from a line on which a gaussian noise is added.
+Faulty points are also added to the data set.
+
+Then, RANSAC is an iterative algorithm for the robust estimation of parameters
+from a subset of inliers from the complete data set. Each iteration
+performs the following tasks:
+
+1. Select ``min_samples`` random samples from the original data and can check
```
Comment (emmanuelle) [inline doc/examples/transform/plot_ransac.py]: can check --> check?

**Comment by emmanuelle** [comment]
@sciunto it doesn't look like you commited the changes suggested by @jni, or did I miss something? I also noted something which is a typo I think, otherwise LGTM.

**Comment by sciunto** [comment]
All is fixed! thanks

**Comment by emmanuelle** [comment]
Merging, thanks.

**Comment by sciunto** [comment]
@emmanuelle   you didn't press the green button, did you? :)

File: doc/examples/transform/plot_ransac.py [hunk 4]
```diff
@@ -4,7 +4,32 @@
 =========================================
 
 In this example we see how to robustly fit a line model to faulty data using
-the RANSAC algorithm.
+the RANSAC (random sample consensus) algorithm.
+
+First, data are generated from a line on which a gaussian noise is added.
```
Comment (soupault) [inline doc/examples/transform/plot_ransac.py]:
```suggestion
Firstly the data are generated by adding a gaussian noise to a linear function. Then, the outlier points are added to the data set.
```

File: doc/examples/transform/plot_ransac.py [hunk 5]
```diff
@@ -4,7 +4,32 @@
...
+Outlier points are also added to the data set.
+
+Then, RANSAC is an iterative algorithm for the robust estimation of parameters
```
Comment (soupault) [inline doc/examples/transform/plot_ransac.py]:
```suggestion
RANSAC iteratively estimates the parameters from the data set. At each iteration the following steps are performed:
```

File: doc/examples/transform/plot_ransac.py [hunk 6]
```diff
@@ -4,7 +4,32 @@
...
+from a subset of inliers from the complete data set. Each iteration
+performs the following tasks:
+
+1. Select ``min_samples`` random samples from the original data and check
+   whether the set of data is valid (see ``is_data_valid`` option).
+2. Estimate a model to the random subset
```
Comment (soupault) [inline doc/examples/transform/plot_ransac.py]:
```suggestion
2. Estimate a model on the random subset
```

File: doc/examples/transform/plot_ransac.py [hunk 7]
```diff
@@ -4,7 +4,32 @@
...
+   (``model_cls.estimate(*data[random_subset]``) and check whether the
+   estimated model is valid (see ``is_model_valid`` option).
+3. Classify all data as inliers or outliers by calculating the residuals
```
Comment (soupault) [inline doc/examples/transform/plot_ransac.py]:
```suggestion
3. Classify all the data points as either inliers or outliers by calculating the residuals using the estimated model ...
```

File: doc/examples/transform/plot_ransac.py [hunk 8]
```diff
@@ -4,7 +4,32 @@
...
+   to the estimated model (``model_cls.residuals(*data)``) - all data samples
+   with residuals smaller than the ``residual_threshold`` are considered as
+   inliers.
+4. Save estimated model as best model if number of inlier samples is
```
Comment (soupault) [inline doc/examples/transform/plot_ransac.py]:
```suggestion
4. If the number of the inlier samples is greater than ever before, save the estimated model as the best model.
```

File: doc/examples/transform/plot_ransac.py [hunk 9]
```diff
@@ -4,7 +4,32 @@
...
+   maximal. In case the current estimated model has the same number of
+   inliers, it is only considered as the best model if it has less sum of
```
Comment (soupault) [inline doc/examples/transform/plot_ransac.py]:
```suggestion
   inliers, it is considered as the best model only if the sum of residuals is lower.
```

**Comment by sciunto** [comment]
Great suggestions. Nice collaborative work. I merged them all @emmanuelle

**Comment by soupault** [comment]
Thanks Francois!

**Pull request merged**